org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/ResourceForIEditorInputFactory.java

```diff
@@ -107,8 +122,15 @@ protected Resource createResource(IStorage storage) throws CoreException {
 	}
 
 	protected Resource createResourceFor(IStorage storage) throws CoreException {
+		String path = null;
+		if (storage.getFullPath() != null)
+			path = storage.getFullPath().toString();
+		if (path == null)
+			path = storage.getName();
+		if (path == null)
+			throw new IllegalArgumentException("Couldn't create EMF Resource for storage " + storage);
 		ResourceSet resourceSet = getResourceSet(storage);
-		URI uri = URI.createPlatformResourceURI(storage.getFullPath().toString(), true);
+		URI uri = URI.createPlatformResourceURI(path, true);
```

Comment:
If we take the name of the storage, we should probably use a different resource scheme. if there is no member in the workspace that matches the name.
@cdietrich WDYT?

Reply:
i did not find the time to investigate/reproduce the issue yet

Reply:
@szarnekow I think there will not be a member in the workspace that matches the name, because without the full path we do not know to which project the storage belongs.
So I would opt for not using the name of the storage.

Reply:
Agreed.